app/models/work_type.rb

```diff
@@ -0,0 +1,127 @@
+  MUSIC = 'music'
+  OTHER = 'other'
+
+  DATA_TYPES = [
+    '3D model', 'Database', 'Documentation', 'Geospatial data', 'Image',
+    'Tabular data', 'Text corpus'
+  ].freeze
+
+  VIDEO_TYPES = [
+    'Conference session', 'Documentary', 'Event', 'Oral history', 'Performance'
+  ].freeze
+
+  SOUND_TYPES = ['Interview', 'Oral history', 'Podcast', 'Speech'].freeze
+
+  TEXT_TYPES = [
+    'Article', 'Capstone', 'Government document', 'Policy brief', 'Preprint', 'Report',
+    'Technical report', 'Thesis', 'Working paper'
+  ].freeze
+
+  SOFTWARE_TYPES = %w[Code Documentation Game].freeze
+
+  IMAGE_TYPES = ['CAD', 'Map', 'Photograph', 'Poster', 'Presentation slides'].freeze
+
+  MUSIC_TYPES = [
+    'Data',
+    'Image',
+    'MIDI',
+    'Musical transcription',
+    'Notated music',
+    'Piano roll',
+    'Software/Code',
+    'Sound',
+    'Text',
+    'Video'
+  ].freeze
+
+  # These types appear below the fold and may be expanded
+  MORE_TYPES = [
+    '3D model', 'Animation', 'Article', 'Book', 'Book chapter', 'Broadcast', 'CAD', 'Capstone',
+    'Code', 'Conference abstract', 'Conference session', 'Correspondence', 'Course/instructional materials',
+    'Data', 'Database', 'Documentary', 'Documentation', 'Dramatic performance',
+    'Essay', 'Ethnography', 'Event', 'Experimental audio/video', 'Field recording',
+    'Game', 'Geospatial data', 'Government document', 'Image', 'Interview',
+    'Journal/periodical issue', 'Manuscript', 'Map', 'MIDI', 'Musical transcription',
+    'Narrative film', 'Notated music', 'Oral history', 'Other spoken word', 'Pamphlet',
+    'Performance', 'Photograph', 'Piano roll', 'Podcast', 'Poetry reading',
+    'Policy brief', 'Poster', 'Portfolio', 'Preprint', 'Presentation recording',
+    'Presentation slides', 'Questionnaire', 'Remote sensing imagery', 'Report',
+    'Software', 'Sound recording', 'Speaker notes', 'Speech', 'Story', 'Syllabus',
+    'Tabular data', 'Technical report', 'Text', 'Text corpus', 'Thesis',
+    'Transcript', 'Unedited recording', 'Video art', 'Video recording',
+    'White paper', 'Working paper'
+  ].freeze
+
+  MIXED_TYPES = (['Music'] + MORE_TYPES).sort.freeze
+
+  attr_reader :id, :label, :html_label, :icon, :cocina_type, :subtypes
+
+  def initialize(**params)
+    @id = params.fetch(:id)
+    @label = params.fetch(:label)
+    @html_label = params.fetch(:html_label)
+    @icon = params.fetch(:icon)
+    @subtypes = params.fetch(:subtypes)
+    @cocina_type = params.fetch(:cocina_type)
+  end
+
+  def self.purl_reservation_type
+    new(id: 'purl_reservation', label: 'PURL reservation', html_label: 'PURL reservation', icon: '', subtypes: [],
+        cocina_type: Cocina::Models::ObjectType.object)
+  end
+
+  def self.find(id)
+    (all + [purl_reservation_type]).find { |work| work.id == id } || raise(InvalidType, "Unknown worktype #{id}")
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  # id is a value acceptable for MODS typeOfResource
```

Comment:
Is this comment still meaningful in the H3 context?

Reply:
Yes, I think it will be. We should be mapping the work types / subtypes in the same way.